browser/brave_ads/device_id/device_id_impl_mac.cc

```diff
@@ -50,15 +50,15 @@ std::string FindBSDNameOfSystemDisk() {
                                                 base::BlockingType::MAY_BLOCK);
 
   struct statfs* mounted_volumes;
-  const int count = getmntinfo(&mounted_volumes, 0);
+  const int count = getmntinfo_r_np(&mounted_volumes, 0);
```

Comment:
The [getmntinfo_r_np](https://keith.github.io/xcode-man-pages/getmntinfo.3.html#getmntinfo_r_np)() function is a thread-safe equivalent of getmntinfo() that allocates a new results buffer on every call and transfers ownership of this buffer to the caller. **The caller is responsible for freeing this memory with free(3)**.

Reply:
@aseren can you please raise an issue, this is only called once but would be great to fix soon. Thanks

Reply:
https://github.com/brave/brave-browser/issues/31458